Modify the login_tty check on Solaris to not include libutil

--- a/meson.build	2020-01-29 11:23:40.831788267 -0800
+++ b/meson.build	2020-01-29 11:31:47.460266960 -0800
@@ -84,8 +84,13 @@
 config_h.set('HAVE_OPENPTY', have_openpty,
              description: 'Define if you have the openpty function.')
 
-config_h.set('HAVE_LOGIN_TTY', util_dep.found() and cc.has_function('login_tty', dependencies: util_dep),
+if host_machine.system().contains('sunos')
+    config_h.set('HAVE_LOGIN_TTY', cc.has_function('login_tty'),
              description: 'Whether login_tty is available')
+else
+    config_h.set('HAVE_LOGIN_TTY', util_dep.found() and cc.has_function('login_tty', dependencies: util_dep),
+             description: 'Whether login_tty is available')
+endif
 
 # headers
 check_headers = [
